includes/Core/Admin/Plugin_Action_Links.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Admin\Plugin_Action_Links
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Admin;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Permissions\Permissions;
+
+/**
+ * Class for managing plugin action links.
+ *
+ * @since 1.41.0
+ * @access private
+ * @ignore
+ */
+class Plugin_Action_Links {
+
+	/**
+	 * Plugin context.
+	 *
+	 * @since 1.41.0
+	 * @var Context
+	 */
+	private $context;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 1.41.0
+	 *
+	 * @param Context $context      Plugin context.
+	 */
+	public function __construct(
+		Context $context
+	) {
+		$this->context = $context;
+	}
+
+	/**
+	 * Registers functionality through WordPress hooks.
+	 *
+	 * @since 1.24.0
+	 */
+	public function register() {
+		add_filter(
+			'plugin_action_links_' . GOOGLESITEKIT_PLUGIN_BASENAME,
+			function ( $links ) {
+
+				if ( current_user_can( Permissions::MANAGE_OPTIONS ) ) {
+					$settings_url = $this->context->admin_url( 'settings' );
+
+					$settings_link = '<a href="' . $settings_url . '">' . __( 'Settings', 'default' ) . '</a>';
+
+					array_unshift( $links, $settings_link );
+				};
+
+				return $links;
+			},
+			10,
+			2
```

Comment:
These two can actually be left off because we only need the first element passed to the filter (`$links`) which is the default. We also don't need a special priority so we omit them when not needed.